container_images/daisy-builder/main.sh

```diff
@@ -21,13 +21,13 @@ WORKFLOW_FILE="$3"      # Workflow to run
 GCS_OUTPUT_BUCKET="$4"  # Destination for artifacts
 
 function generate_new_version() {
-  local VERSION=`/versiongenerator --token-file-path=${GITHUB_ACCESS_TOKEN} --org=${REPO_OWNER} --repo=${REPO_NAME}`
+  local VERSION_OUT=`/versiongenerator --token-file-path=${GITHUB_ACCESS_TOKEN} --org=${REPO_OWNER} --repo=${REPO_NAME}`
 
   if [[ $? -ne 0 ]]; then
```

Comment:
$? now will refer to the success of the VERSION_OUT= assignment, and will probably always be true.

Reply:
changed it to catch the error code correctly

Reply:
Sorry, I wasn't clear. $? will be set to the value of the assignment, which is typically the value of the right operand. But this statement is actually not an assignment, but a `local`. This form of variable initialization always returns 0.

Reply:
Consider changing to:
```
local VERSION_OUT
if ! VERSION_OUT=`/versiongenerator --token-file-path=${GITHUB_ACCESS_TOKEN} --org=${REPO_OWNER} --repo=${REPO_NAME} 2>&1`; then
    echo "could not generate version: ${VERSION_OUT}"
    return 1
fi
echo $VERSION_OUT
```

Reply:
changed